src/app/courses-load-more/courses-load-more.component.ts

```diff
@@ -11,22 +14,20 @@ export class CoursesLoadMoreComponent implements OnInit {
 	@Input()
 	public listCourses: Course[] = [];
 
-	constructor() {
+	private coursesService: CoursesService;
+
+	constructor(coursesService: CoursesService) {
+		this.coursesService = coursesService;
 	}
 
 	ngOnInit() {
 	}
 
 	onLoadMore() {
 		this.showLoader = true;
-			const id: number = this.listCourses.length + 1;
-			this.listCourses.push(new Course({
-				id: id,
-				title: 'demo Title ' + id,
-				creationDate: '10-10-2019',
-				duration: 15,
-				description: 'demo Description ' + id,
-			}));
-			this.showLoader = false
+		this.coursesService.getList().then((items: ICourse[]) => {
+			map(items, (item: CourseProps) => this.listCourses.push(new Course(item)));
+		});
+		this.showLoader = false
```

Comment:
You should do this action in `then`. Otherwise there's no delay between showing and hiding of this loader